### PR TITLE
HBASE-22930 Set unique name to longCompactions/shortCompactions threads

### PR DESCRIPTION
**Pull request by pankaj72981**
Changes for branch-1.

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java [hunk 1]
```diff
@@ -131,11 +132,12 @@ public Thread newThread(Runnable r) {
   private void createSplitExcecutors() {
     final String n = Thread.currentThread().getName();
     int splitThreads = conf.getInt(SPLIT_THREADS, SPLIT_THREADS_DEFAULT);
+    final AtomicInteger splitThreadCounter = new AtomicInteger(0);
     this.splits =
         (ThreadPoolExecutor) Executors.newFixedThreadPool(splitThreads, new ThreadFactory() {
           @Override
```
Comment (Reidddddd) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java]: Only one place not sure: shouldn't the counter be placed in `ThreadFactory`?

Comment (anoopsjohn) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java]: +1 on what Reid said.

Comment (anoopsjohn) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java]: And same way below also.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 56s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| :green_heart: |  mvninstall  |   8m 26s |  branch-1 passed  |
| :green_heart: |  compile  |   2m 20s |  branch-1 passed with JDK v1.8.0_222  |
| :broken_heart: |  compile  |   0m 22s |  hbase-server in branch-1 failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   1m 45s |  branch-1 passed  |
| :green_heart: |  shadedjars  |   3m 11s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 37s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 43s |  branch-1 passed with JDK v1.7.0_232  |
| :blue_heart: |  spotbugs  |   3m  6s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   3m  2s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   2m 10s |  the patch passed  |
| :green_heart: |  compile  |   2m  5s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javac  |   2m  5s |  the patch passed  |
| :broken_heart: |  compile  |   0m 24s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 24s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   1m 43s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   3m 19s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |   5m 44s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 49s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  findbugs  |   3m 35s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  | 151m 41s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   | 198m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/645 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7cc6dc835d2f 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-645/out/precommit/personality/provided.sh |
| git revision | branch-1 / 5c4d8e0 |
| Default Java | 1.7.0_232 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_222 /usr/lib/jvm/zulu-7-amd64:1.7.0_232 |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/artifact/out/branch-compile-hbase-server-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/testReport/ |
| Max. process+thread count | 5035 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java (hunk 1 above)
Comment (pankaj72981) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplitThread.java]: Thanks @Reidddddd and @anoopsjohn  for the quick review, have moved the counter local to  ThreadFactory in commit-2.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 17s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| :green_heart: |  mvninstall  |   8m 11s |  branch-1 passed  |
| :green_heart: |  compile  |   2m  3s |  branch-1 passed with JDK v1.8.0_222  |
| :broken_heart: |  compile  |   0m 21s |  hbase-server in branch-1 failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   1m 45s |  branch-1 passed  |
| :green_heart: |  shadedjars  |   3m 15s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 42s |  branch-1 passed with JDK v1.7.0_232  |
| :blue_heart: |  spotbugs  |   3m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   2m 58s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   2m  7s |  the patch passed  |
| :green_heart: |  compile  |   1m 56s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javac  |   1m 56s |  the patch passed  |
| :broken_heart: |  compile  |   0m 21s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 21s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   1m 41s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   3m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |   5m 15s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 42s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  findbugs  |   3m 12s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  | 159m 23s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   | 203m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/645 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 665c5839d545 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-645/out/precommit/personality/provided.sh |
| git revision | branch-1 / b5b871c |
| Default Java | 1.7.0_232 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_222 /usr/lib/jvm/zulu-7-amd64:1.7.0_232 |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/artifact/out/branch-compile-hbase-server-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/testReport/ |
| Max. process+thread count | 4862 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
@pankaj72981  You have approvals here sir but need to rebase?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   2m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| :green_heart: |  mvninstall  |  10m 47s |  branch-1 passed  |
| :green_heart: |  compile  |   0m 53s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  compile  |   0m 56s |  branch-1 passed with JDK v1.7.0_232  |
| :green_heart: |  checkstyle  |   1m 54s |  branch-1 passed  |
| :green_heart: |  shadedjars  |   3m 29s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 38s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK v1.7.0_232  |
| :blue_heart: |  spotbugs  |   3m  3s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   2m 59s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   2m  5s |  the patch passed  |
| :green_heart: |  compile  |   0m 42s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javac  |   0m 42s |  the patch passed  |
| :green_heart: |  compile  |   0m 46s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  javac  |   0m 46s |  the patch passed  |
| :green_heart: |  checkstyle  |   1m 43s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   3m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |   5m 12s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| :green_heart: |  javadoc  |   0m 30s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   0m 42s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  findbugs  |   3m  4s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  | 141m 17s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   0m 28s |  The patch does not generate ASF License warnings.  |
|  |   | 187m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/645 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux a68aa31fda1f 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-645/out/precommit/personality/provided.sh |
| git revision | branch-1 / daf1faa |
| Default Java | 1.7.0_232 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_222 /usr/lib/jvm/zulu-7-amd64:1.7.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/5/testReport/ |
| Max. process+thread count | 4654 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-645/5/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**